wandb/integration/keras/keras.py

```diff
@@ -642,58 +701,19 @@ def _log_weights(self):
         return metrics
 
     def _log_gradients(self):
-        if not self.training_data:
-            raise ValueError("Need to pass in training data if logging gradients")
-
-        X_train = self.training_data[0]
-        y_train = self.training_data[1]
+        self._grad_accumulator_model.fit(
+            self._training_data_x,
+            self._training_data_y,
+            verbose=0,
+            callbacks=[self._grad_accumulator_callback],
```

Comment:
Does this do an entire epoch by default?  The previous implementation just passed the data through and extracted the gradients.  I'm not super clear on what's happening with our custom optimizer but I wonder if we could get this data more efficiently.

Reply:
That was causing OOM. Here we are basically reusing keras's fit logic (so generators, tf.Dataset etc work out of the box). Extracting gradients batch by batch manually (using `tf.GradientTape`) also causes OOM.